packages/backend/src/graphql/mutations/create-flow-transfer.ts

```diff
@@ -13,15 +14,46 @@ const createFlowTransfer = async (
   params: Params,
   context: Context,
 ) => {
-  const { flowId, newOwnerId } = params.input
-  const transfer: FlowTransfer = await context.currentUser
+  const { flowId, newOwnerEmail } = params.input
+
+  const newOwner = await User.query()
+    .findOne({
+      email: newOwnerEmail,
+    })
+    .throwIfNotFound({
+      message:
+        'User email does not exist on Plumber, please type in an email account of a user who has logged in before!',
+    })
+
+  // don't allow transferring of pipe to oneself
+  if (context.currentUser.id === newOwner.id) {
+    throw new Error(
+      'You cannot transfer the pipe to yourself, please type in another email',
+    )
+  }
+
+  // check for existing pending transfer to avoid duplicates
+  const existingTransfer: FlowTransfer = await FlowTransfer.query().findOne({
+    flow_id: flowId,
+    old_owner_id: context.currentUser.id,
+    new_owner_id: newOwner.id,
+    status: 'pending',
+  })
```

Comment:
2 comments - one related to concurrency, and one nit.

**Concurrency**: nice that you're checking this! But typically, existence checks based on querying will fail if there's an concurrent query. E.g. assume 2 concurrent queries, your 2 selects may run before your 1st insert runs.

The typical way to solve this is via a partial unique index; this also allows you to omit doing your initial select query. Something like:
```
// Enforce unique flow IDs for pending transfers
CREATE UNIQUE INDEX unique_transfers
ON flow_transfers (flow_id)
WHERE status = 'pending'
```

To be fair, we don't really get concurrent queries, but I feel like indexes are cheap at our scale so might as well add it(@pregnantboy may disagree because of over-engineering?)

**Nit**: For existence queries, you can use `resultSize` instead of grabbing the entire row. E.g:
```
// Assume no concurrency
const hasExistingTransfer =
  (await FlowTransfer.query()
    .findOne({
      flow_id: flowId,
      old_owner_id: context.currentUser.id,
      new_owner_id: newOwner.id,
      status: 'pending',
    })
    .resultSize()) === 1
```
But to be really honest i've not verified if objection optimizes `resultSize` into a `COUNT` query...